File: src/main/java/com/todolist/task/service/adapter/http/TaskController.java
package com.todolist.task.service.adapter.http;

import static com.todolist.task.service.common.Constant.USER_ID;

import com.todolist.task.service.adapter.http.models.CreateTaskCommand;
import com.todolist.task.service.application.TaskApplicationService;
import com.todolist.task.service.application.models.TaskDTO;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/task")
@RequiredArgsConstructor
public class TaskController {

  private final TaskApplicationService taskApplicationService;

  @PostMapping
  public void createTask(@RequestHeader(USER_ID) String user,
      @RequestBody CreateTaskCommand createTaskCommand) {
    taskApplicationService.createTask(user, createTaskCommand);
  }

  @GetMapping
  public List<TaskDTO> getAllTasks(@RequestHeader(USER_ID) String user) {
    return taskApplicationService.getAllTasks(user);
  }

}

File: src/main/kotlin/com/todolist/task/service/adapter/http/TaskController.kt
package com.todolist.task.service.adapter.http

import com.todolist.task.service.application.models.CreateTaskCommand
import com.todolist.task.service.application.TaskApplicationService
import com.todolist.task.service.application.models.TaskDTO
import com.todolist.task.service.common.Constant
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping
class TaskController(
    private val taskApplicationService: TaskApplicationService
) {
    @PostMapping("/task")
    fun createTask(
        @RequestHeader(Constant.USER_ID) user: String,
        @RequestBody createTaskCommand: CreateTaskCommand
    ) {
        taskApplicationService.createTask(user, createTaskCommand)
    }

    @GetMapping("/tasks")
    fun getAllTasks(@RequestHeader(Constant.USER_ID) user: String): List<TaskDTO>? {
        return taskApplicationService.getAllTasks(user)
    }

    @DeleteMapping("/task/{id}")
    fun deleteTask(
        @PathVariable("id") taskId: String,
        @RequestHeader(Constant.USER_ID) user: String
    ) {
        taskApplicationService.deleteTask(taskId, user)
    }
}
